enterprise/coderd/prebuilds/reconcile.go

```diff
@@ -87,16 +88,27 @@ func (c *StoreReconciler) Run(ctx context.Context) {
 		slog.F("backoff_interval", c.cfg.ReconciliationBackoffInterval.String()),
 		slog.F("backoff_lookback", c.cfg.ReconciliationBackoffLookback.String()))
 
+	var wg sync.WaitGroup
 	ticker := c.clock.NewTicker(reconciliationInterval)
 	defer ticker.Stop()
 	defer func() {
 		c.done <- struct{}{}
+		wg.Wait()
```

Comment:
Shouldn't `wg.Wait()` be called _before_ writing to `c.done`? 

```suggestion
		wg.Wait()
		c.done <- struct{}{}
```